content/automate/ManageIQ/Service/Retirement/StateMachines/ServiceRetirement.class/__methods__/update_serviceretirement_status.rb

```diff
@@ -0,0 +1,58 @@
+module ManageIQ
+  module Automate
+    module Service
+      module Retirement
+        module StateMachines
+          module ServiceRetirement
+            class UpdateServiceRetirementStatus
+
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                task = @handle.root['service_retire_task']
+
+                if task.nil?
```

Comment:
@d-m-u Normally, this would be correct in that we would raise if an object we expected to be present was missing. In this case though, we want to be backward compatible, so you can remove the the if task.nil? code block.